unittests/multi_clock/gen.py

```diff
@@ -35,11 +34,11 @@ def main():
     m.bind_name('dt_req_next', if_(m.prev_clk_val, m.t_lo, m.t_hi))
 
     # increment the time request
-    m.bind_name('dt_req_incr', m.dt_req + m.neg_emu_dt)
+    m.bind_name('dt_req_incr', m.dt_req - m.emu_dt)
 
     # determine the next period
     m.bind_name('dt_req_imm', if_(m.req_grant, m.dt_req_next, m.dt_req_incr))
-    m.set_next_cycle(m.dt_req, m.dt_req_imm[31:0], clk=m.emu_clk, rst=m.emu_rst)
+    m.set_next_cycle(m.dt_req, m.dt_req_imm, clk=m.emu_clk, rst=m.emu_rst, check_format=False)
```

Comment:
``check_format=False`` is needed here because ``dt_req_imm`` doesn't have the same format as ``dt_req``.